### ISSUE TITLE
swingjs does not respect the scope of a final variable

### ISSUE THREAD

**Issue by warownia1** [BobHanson/java2script#205]
When the final variable is used by the inner runnable which is then run outside of the scope, the variable value may change during execution.
Example:
~~~java
  static ExecutorService executor = Executors.newSingleThreadScheduledExecutor();
  public static void main(String[] args) {
    final int N = 10;
    
    Runnable r = () -> {
      for (int k = 0; k < 5; k++)
      {
        final int i = k;
        executor.submit(() -> {
          System.out.println(i);
        });
      }
    };
    r.run();
  }
~~~

Expected:
~~~
0
1
2
3
4
~~~
Got:
~~~
4
4
4
4
4
~~~
This example also demonstrated the issue #204 

Additionally, moving executor declaration inside the main method causes `TypeError: this.$finals$ is undefined` when run.

**Comment by BobHanson** [comment]
that's "final"?

**Comment by BobHanson** [comment]
Hmm, so the delay is picking up the final value. I suppose it would. There might not be anything I can do about that.

**Comment by BobHanson** [comment]
It's the issue of Runnable in a Runnable. OK...

**Comment by BobHanson** [comment]
Thank you for that simple test! The test is integrated into test/Test_Final.java   passing all tests. Issue with HashTable design for following final values using class binary name, but then Runnable within Runnable doesn't work correctly. Fixed and pushed as java2script 3.3.1-v3.

**Comment by BobHanson** [comment]
Just pointing out that this was a transpiler fix. Java11 transpiler hasn't been created, though.
